### PR TITLE
Improvement gke cluster creation

### LINKED ISSUE #4786: Improvement to GKE External Cluster
**What would you like to be added? / User Story**:
Priority: high
- [ ] Put a title `Node Pool Setting`  
- [ ] default name for NodePool `default-pool`
- [ ] Show somehow that k8s version for the nodepool is same as cluster control plane version
- [ ] Move `Node Count` under this , make it consistent for all providers, either `Replicas` (same as KKP) or `Node Count`
-  [ ] Create MD: 
        -  Lets remove  `Description`, reason is for some sizes the size doesn't fit the screen
        - Show name, memory and vcpus

Below point is on priority: normal
- [ ] Create Cluster:     
      - [ ] Provide fields for machinetype, disktype and disksizegb
      -  `memory` for Memory in MB
      payload:
      // default 25 Gb, cannnot be less that 10Gb
      gkeclusterSpec.nodeConfig.diskSizeGb
      gkeclusterSpec.nodeConfig.diskType
      gkeclusterSpec.nodeConfig.machineType

      List object will look like this:
      `{
          "name": "a2-highgpu-8g",
          "description": "Accelerator Optimized: 8 NVIDIA Tesla A100 GPUs, 96 vCPUs, 680GB RAM",
          "memory": 696320,
          "vcpus": 96
        },`
     
** Should look like: a2-highgpu-8g, 96 vCPUs, 696320MB RAM

### PR DESCRIPTION
**Pull request by ahmadhamzh**
**What this PR does / why we need it**:
refactor the input fields and add the machine and disk types for GKE cluster creation
**Which issue(s) this PR fixes**:
Fixes #4786

**What type of PR is this?**
/kind feature

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
add the machine types and disk types fields for GKE cluster creation 
```
```documentation
NONE
```

### CODE REVIEW

File: src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts [hunk 1]
```diff
@@ -224,6 +245,13 @@ export class GKEClusterSettingsComponent
         .subscribe(_ => {
           this._getGKEKubernetesVersions();
         });
+      this.control(Controls.Zone)
+        .valueChanges.pipe(takeUntil(this._unsubscribe))
```
Comment (KhizerRehan) [inline src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts:249]: Try to clear disktypes and machine types when new API call is send.

Check `_clearVmSize` in AKSClusterSettingsComponent as a reference

File: src/app/external-cluster-wizard/steps/external-cluster/provider/gke/template.html [hunk 2]
```diff
@@ -72,35 +107,55 @@
...
-    <km-number-stepper *ngIf="isDialogView()"
-                       hint="Boot disk size is permanent."
+    <km-combobox filterBy="name"
+                 inputLabel="Select Machine Type..."
+                 hint="Select vCPU cores and memory"
+                 [label]="machineTypeLabel"
+                 [options]="machineTypes"
+                 [formControlName]="Controls.MachineTypes">
+      <div *option="let machineType">
+        {{machineType.name+" ("+machineType.vcpus +" vCPUs "+machineType.memory+"MB RAM)"}}
```
Comment (KhizerRehan) [inline src/app/external-cluster-wizard/steps/external-cluster/provider/gke/template.html:136]: **Changes should be tested on both (Wizard/MD dialog)**

Don't add computation on template instead create a method in `ts` file which can be also passed as [valueFormatter] in component. 


Show both name + computation on Selection Too

machine-type (display-name)

File: src/app/external-cluster-wizard/steps/external-cluster/provider/gke/template.html [hunk 3]
```diff
@@ -72,35 +107,55 @@
...
+    </km-combobox>
+
+    <km-combobox filterBy="name"
+                 inputLabel="Select Disk Type..."
+                 hint="SSD persistent disk is better for random IOPS or for streaming throughput with low latency."
+                 [label]="diskTypeLabel"
+                 [options]="diskTypes"
+                 [formControlName]="Controls.DiskTypes">
+      <div *option="let diskType">
+        {{diskType.description}}
```
Comment (KhizerRehan) [inline src/app/external-cluster-wizard/steps/external-cluster/provider/gke/template.html:147]: **Changes should be tested on both (Wizard/MD dialog)**

disk-type (display-name)

**show as**
showas

File: src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts [hunk 4]
```diff
@@ -234,11 +262,49 @@ export class GKEClusterSettingsComponent
...
+            this.control(Controls.DiskTypes).setValue(diskType.name);
+          }
+        });
+        this.diskTypes = diskTypes;
+        this.diskTypeLabel = this.diskTypes?.length ? DiskTypeState.Ready : DiskTypeState.Empty;
+      });
+  }
+
+  private _getGKEMachineTypes(): void {
+    this._externalClusterService
```
Comment (KhizerRehan) [inline src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts:281]: correct  state shown based on API calls see method `_getGKEMachineTypesForMachineDeployment`
set value before API call
this.<LabelName> = <Type>.Loading;

**Review by KhizerRehan — CHANGES_REQUESTED** [review]
Some feedback added

File: src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts [hunk 5]
```diff
@@ -234,11 +262,49 @@ export class GKEClusterSettingsComponent
...
+  }
+
   private _getGKEDiskTypesForMachineDeployment(): void {
+    this.diskTypeLabel = DiskTypeState.Loading;
     this._externalMachineDeploymentService
       .getGKEDiskTypesForMachineDeployment(this.projectID, this.cluster.id)
-      .pipe(map((diskTypes: GCPDiskType[]) => diskTypes.map(type => type.name + ` (${type.description})`)))
-      .subscribe((diskTypes: string[]) => (this.diskTypes = diskTypes));
+      .subscribe((diskTypes: GCPDiskType[]) => {
+        diskTypes.map(diskType => {
```
Comment (KhizerRehan) [inline src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts:300]: Check this comment: https://github.com/kubermatic/dashboard/pull/4917#discussion_r960396002

File: src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts [hunk 6]
```diff
@@ -280,17 +346,19 @@ export class GKEClusterSettingsComponent
...
-        map((machineSizes: GCPMachineSize[]) =>
-          machineSizes.map((machineType: GCPMachineSize) => {
-            return `${machineType.name} ${machineType.description})`;
-          })
-        )
-      )
-      .subscribe((machineTypes: string[]) => (this.machineTypes = machineTypes));
+      .getGKEMachineTypesForMachineDeployment(this.projectID, this.cluster.id)
+      .subscribe((machineTypes: GCPMachineSize[]) => {
+        machineTypes.map(machineType => {
```
Comment (KhizerRehan) [inline src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts:354]: Check this comment: https://github.com/kubermatic/dashboard/pull/4917#discussion_r960396002

File: src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts [hunk 7]
```diff
@@ -234,11 +262,49 @@ export class GKEClusterSettingsComponent
...
+        this.diskTypeLabel = this.diskTypes?.length ? DiskTypeState.Ready : DiskTypeState.Empty;
+      });
+  }
+
+  private _getGKEMachineTypes(): void {
+    this._externalClusterService
+      .getGKEMachineTypes(this.controlValue(Controls.Zone))
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe((machineTypes: GCPMachineSize[]) => {
+        machineTypes.map(machineType => {
```
Comment (KhizerRehan) [inline src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts:285]: Check this comment: https://github.com/kubermatic/dashboard/pull/4917#discussion_r960396002

**Review by KhizerRehan — CHANGES_REQUESTED** [review]

File: src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts [hunk 8]
```diff
@@ -234,11 +262,49 @@ export class GKEClusterSettingsComponent
       .subscribe((zones: string[]) => (this.zones = zones));
   }
 
+  private _getGKEDiskTypes(): void {
+    this._externalClusterService
+      .getGKEDiskTypes(this.controlValue(Controls.Zone))
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe((diskTypes: GCPDiskType[]) => {
+        diskTypes.map(diskType => {
+          if (diskType.name === this.DISK_TYPE_DEFAULT_VALUE) {
```
[the remark this answers is not in the document]
Comment (ahmadhamzh) [reply src/app/external-cluster-wizard/steps/external-cluster/provider/gke/component.ts:271]: can we use `includes` here ?? `diskTypes` is an array of objects and `this.DEFAULT_LOCATION` is a string

**Comment by KhizerRehan** [comment]
@ahmadhamzh please test also
- once you select go to details step
- go back change preset 
- select another valid preset `next` button is still `disabled` currently. It should be enabled

**Pull request merged**